src/highdicom/content.py

```diff
@@ -936,3 +1067,536 @@ def __init__(
+            bits_allocated
+        ]
+
+        if lut_explanation is not None:
+            _check_long_string(lut_explanation)
+            self.LUTExplanation = lut_explanation
+
+    @property
+    def lut_data(self) -> np.ndarray:
+        """np.ndarray: LUT data formatted as np.ndarray."""
+        if self.bits_allocated == 8:
+            np_dtype = np.uint8
+        elif self.bits_allocated == 16:
+            np_dtype = np.uint16
+        else:
+            raise RuntimeError("Invalid LUT descriptor.")
+        return np.frombuffer(self.LUTData, np_dtype)
+
+    @property
+    def first_mapped_value(self) -> int:
+        """int: Pixel value that will be mapped to the first value in the
+        LUT.
+        """
+        return self.LUTDescriptor[1]
+
+    @property
+    def bits_allocated(self) -> int:
+        """int: Bits allocated for the LUT data. 8 or 16."""
+        return self.LUTDescriptor[2]
+
+
+class ModalityLUT(LUT):
+
+    """Dataset describing a modality lookup table."""
+
+    def __init__(
+        self,
+        modality_lut_type: Union[RescaleTypeValues, str],
+        first_mapped_value: int,
+        lut_data: np.ndarray,
+        lut_explanation: Optional[str] = None
+    ):
+        """
+
+        Parameters
+        ----------
+         modality_lut_type: Union[highdicom.RescaleTypeValues, str]
+            String or enumerated value specifying the units of the output of
+            the LUT operation.
+          first_mapped_value: int
+            Pixel value that will be mapped to the first value in the
+            lookup-table.
+        lut_data: np.ndarray
+            Lookup table data. Must be of type uint8 or uint16.
+        lut_explanation: Union[str, None], optional
+            Free-form text explanation of the meaning of the LUT.
+
+        Note
+        ----
+        After the LUT is applied, a pixel in the image with value equal to
+        ``first_mapped_value`` is mapped to an output value of ``lut_data[0]``,
+        an input value of ``first_mapped_value + 1`` is mapped to
+        ``lut_data[1]``, and so on.
+
+        """
+        super().__init__(
+            first_mapped_value=first_mapped_value,
+            lut_data=lut_data,
+            lut_explanation=lut_explanation
+        )
+        if isinstance(modality_lut_type, RescaleTypeValues):
+            self.ModalityLUTType = modality_lut_type.value
+        else:
+            _check_long_string(modality_lut_type)
+            self.ModalityLUTType = modality_lut_type
+
+
+class VOILUT(Dataset):
+
+    """Dataset describing a value-of-interest lookup table."""
+
+    def __init__(
+        self,
+        window_center: Union[float, Sequence[float], None] = None,
+        window_width: Union[float, Sequence[float], None] = None,
+        window_explanation: Union[str, Sequence[str], None] = None,
+        voi_lut_function: Union[VOILUTFunctionValues, str, None] = None,
+        voi_luts: Optional[Sequence[LUT]] = None,
```

Comment:
I think the confusion comes from the fact that the `VOILUT` class corresponds to the VOI LUT module rather than an item of the VOI LUT Sequence attribute. So far, we have implemented classes for sequence items, but not entire modules, and I would stick to that approach for consistency.

Reply:
Unfortunately we have inherited quite a lot of confusion from the standard here. The primary point of confusion for both VOI LUT and Modality LUT is that "LUT" is used to refer to both:
- The general Modality/VOI "transformation", regardless of whether that transformation is actually represented by a LUT or a rescaling/windowing (in which no LUT is involved at all), and
- The actual LUT that *may* implement the transformation

E.g. a windowing transformation still belongs to the VOI LUT even though there is no actual LUT involved.

I have stuck to the terminology in the standard pretty closely, even though I think it is highly misleading. I would not be opposed to renaming classes to more accurately reflect what they do, for example "VOITransform" and "ModalityTransform". On balance I think sticking with the standard is probably going to cause less confusion but it's a close call in my opinion.

I have renamed the parameter to `luts` in eba0dd2 in both the `VOILUT` and `SoftcopyVOILUT` classes as you suggest to try and clear this up a little bit.

When I implemented the `VOILUT` class, my intent was that it could be used directly for items of the [FrameVOILUTSequence](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.7.6.16.2.html#table_C.7.6.16-11b) as part of the multi frame functional groups for IODs such as `ParametricMap` (though as mentioned somewhere else I stopped short of actually implementing that change to `ParametricMap` in this PR for fear of it becoming too large in scope) and potentially other multi-frame images in the future and simultaneously be a base class for `SoftcopyVOILUT` to re-use the same code as much as possible. So it's not intended *just* as an abstract base class. See your suggestion [here](https://github.com/herrmannlab/highdicom/pull/139#pullrequestreview-887937422). Therefore I would argue that we should keep it as a class as part of the public API so that `ParametricMap` may use it in the future.

So overall I think we should stick with the current structure, but potentially give things clearer names. If you want to discuss further I suggest we talk in person to iron it out.

Reply:
Yes, let's chat..